packages/babel-parser/src/plugins/flow/index.ts

```diff
@@ -2181,7 +2183,9 @@ export default (superClass: typeof Parser) =>
     ): T {
       let result: T;
       if (this.state.noArrowParamsConversionAt.includes(node.start)) {
-        this.state.noArrowParamsConversionAt.push(this.state.start);
+        this.state.noArrowParamsConversionAt.push(
+          this.state.sourceToOffsetPos(this.state.start),
```

Comment:
In general, what should be the rule for when to use `sourceToOffsetPos` when interacting with `state.start`, and when is it not needed?

Reply:
For these places where we store indexes I was mostly just trying to see if the majority of comparisons were against the source or the offset (ie comparing against a nodes indexes, or parser state indexes).

Maybe it makes more sense to always use parser state indexes on the `this.state` though?
I'm kind of thinking that now looking at it. I can go ahead and make that change if you agree.

Reply:
I went ahead and updated it to be the way I described (every position property stored on the parser state is now always relative to the parser pos).

Reply:
Thanks :)